packages/date/src/index.js

```diff
@@ -440,4 +463,49 @@ export function getDate( dateString ) {
 	return momentLib.tz( dateString, WP_ZONE ).toDate();
 }
 
+/**
+ * Creates a moment instance using the given timezone or, if none is provided, using global settings.
+ *
+ * @param {(Date|string|Moment|null)}   dateValue  Date object or string, parsable
+ *                                                 by moment.js.
+ * @param {(string|number|null)}        timezone   Timezone to output result in or
+ *                                                 a UTC offset.
+ *                                                 Defaults to timezone from site.
+ *                                                 See momentjs.com.
+ * @return {Moment}  a moment instance.
+ */
+function buildMoment( dateValue, timezone ) {
+	if ( timezone && ! isUtcOffset( timezone ) ) {
+		return momentLib( dateValue ).tz( timezone );
+	}
+
+	if ( timezone && isUtcOffset( timezone ) ) {
+		return momentLib( dateValue ).utcOffset( timezone );
+	}
+
+	if ( settings.timezone.string ) {
+		return momentLib( dateValue ).tz( settings.timezone.string );
+	}
+
+	return momentLib( dateValue ).utcOffset( settings.timezone.offset );
+}
+
+const VALID_UTC_OFFSET = /^[+-][0-1][0-9](:[0-9][0-9])?$/;
```

Comment:
Ideally top of file, and documented, and including a reference to any applicable specification or reference from which this is inspired.

>In almost all cases, a constant should be defined in the top-most scope of a file.

https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#constants

(Technically it would be correct to say this is "top-most scope", but as far as placement, conventionally it would be expected to be declared toward the top)

Reply:
As requested, I moved the regexp to the top of file and added a reference to [Wikipedia’s article on UTC time offsets](https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC).

Fixed in c3d5fe4.